hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvRowProjector.java

```diff
@@ -92,6 +93,21 @@
         return extractors;
     }
 
+    public Object[] projectData(Data key, Data value) {
+        keyTarget.setTarget(null, key);
```

Comment:
@gierlachg This does not seem to play well with all the QueryTargets. Some QueryTargets (`AvroQueryTarget`  and `JsonQueryTarget` ) expects the second parameter as null. 
I wonder why there was no test failure. Are tests missing or we never expect `Avro/JsonQueryTarget` here ?